linodecli/output.py

```diff
@@ -173,29 +167,48 @@ def _select_json_elements(self, keys, json_res):
             if k in keys:
                 ret[k] = v
             elif isinstance(v, dict):
-                v = self._select_json_elements(keys, v)
+                v = OutputHandler._select_json_elements(keys, v)
                 if v:
                     ret[k] = v
         return ret
 
-    def _markdown_output(self, header, data, columns):
+    def _markdown_output(self, header, data, columns, to):
         """
         Pretty-prints data in a Markdown-formatted table.  This uses github's
         flavor of Markdown
         """
+        content = self._build_output_content(
+            data, columns,
+            value_transform=lambda attr, v: attr.render_value(v, colorize=False))
+
+        if header:
+            print("| " + " | ".join([str(c) for c in header]) + " |", file=to)
+            print("|---" * len(header) + "|", file=to)
+
+        for row in content:
+            print("| " + " | ".join([str(c) for c in row]) + " |", file=to)
+
+    def _build_output_content(
+            self,
+            data,
+            columns,
+            header=None,
+            value_transform=lambda attr, model: model):
+        """
+        Returns the `content` to be displayed by the corresponding output function.
+        `value_transform` allows functions to specify how each value should be formatted.
+        """
+
         content = []
 
+        if self.headers and header is not None:
```

Comment:
I think `self.headers` is almost equivalent to `self.headers and header is not None` here unless the default behavior is overridden by the magic method. Is that the case?